test/unit/lib/population_clone_job_test.rb

```diff
@@ -14,6 +14,105 @@ def test_perform_full_deck
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = ['CPT']
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should be from the CPT code system
+    assert_equal 'CPT', cloned_procedure_codes[0].codeSystem
+  end
+
+  def test_preferred_code_procedure_with_vendor_perference_not_found
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = ['FAKECS']
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should not be from the FAKECS code system
+    assert_not_equal 'FAKECS', cloned_procedure_codes[0].codeSystem
+  end
+
+  def test_preferred_code_procedure_with_ordered_vendor_perference_snomedct_first
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = %w[SNOMEDCT CPT]
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should be from the SNOMEDCT code system
+    assert_equal 'SNOMEDCT', cloned_procedure_codes[0].codeSystem
+  end
+
+  def test_preferred_code_procedure_with_ordered_vendor_perference_cpt_first
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = %w[CPT SNOMEDCT]
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should be from the SNOMEDCT code system
+    assert_equal 'CPT', cloned_procedure_codes[0].codeSystem
+  end
+
```

Comment:
These two functions are also almost identical except for the specific code types.